src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -30,7 +30,51 @@ internal RuntimeType GetTypeChecked() =>
         /// </summary>
         /// <param name="value">An IntPtr handle to a RuntimeType to create a <see cref="RuntimeTypeHandle"/> object from.</param>
         /// <returns>A new <see cref="RuntimeTypeHandle"/> object that corresponds to the value parameter.</returns>
-        public static RuntimeTypeHandle FromIntPtr(IntPtr value) => new RuntimeTypeHandle(Type.GetTypeFromHandleUnsafe(value));
+        public static RuntimeTypeHandle FromIntPtr(IntPtr value) => new RuntimeTypeHandle(GetTypeFromHandle(value));
+
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "RuntimeTypeHandle_GetTypeObjectFromHandleSlow")]
+        private static partial void GetTypeObjectFromHandleSlow(
+            IntPtr handle,
+            ObjectHandleOnStack typeObject);
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        private static extern RuntimeType? GetTypeObjectFromHandleInternal(IntPtr handle);
```

Comment:
```suggestion
        private static extern RuntimeType? GetTypeObjectFromHandleIfExist(IntPtr handle);
```
Existing naming art `GetAssemblyIfExists`